launch/launcher_test.go

```diff
@@ -219,351 +131,409 @@ func testLauncher(t *testing.T, when spec.G, it spec.S) {
-							t.Fatalf("stdout was empty: stderr: %s\n", stderr)
-						}
-						if diff := cmp.Diff(strings.ReplaceAll(stdout, "\r\n", "\n"), "hi from app\nprof2,prof2start,prof1,prof1start,\n"); diff != "" {
-							t.Fatalf("syscall.Exec stdout did not match: (-got +want)\n%s\n", diff)
-						}
+							if shell.process.Script {
+								t.Fatalf("did not expect script process")
+							}
+						})
+					})
+
+					when("has no args", func() {
+						it.Before(func() {
+							process.Args = []string{}
+						})
+
+						when("linux", func() {
+							it.Before(func() {
+								h.SkipIf(t, runtime.GOOS == "windows", "windows test")
+							})
+
+							it("is script", func() {
+								h.AssertNil(t, launcher.LaunchProcess("/path/to/launcher", process))
+								h.AssertEq(t, shell.nCalls, 1)
+
+								if !shell.process.Script {
+									t.Fatalf("expected script process")
+								}
+							})
+						})
+
+						when("windows", func() {
+							it.Before(func() {
+								h.SkipIf(t, runtime.GOOS != "windows", "windows test")
+							})
+
+							it("is not script", func() {
+								h.AssertNil(t, launcher.LaunchProcess("/path/to/launcher", process))
+								h.AssertEq(t, shell.nCalls, 1)
+
+								if shell.process.Script {
+									t.Fatalf("did not expect script process")
+								}
+							})
+						})
 					})
 				})
 
-				when("app has '.profile'", func() {
+				when("buildpack API < 0.4", func() {
 					it.Before(func() {
-						if runtime.GOOS == "windows" {
-							mkfile(t, "set OUT=%OUT%profile", filepath.Join(tmpDir, "launch", "app", ".profile.bat"))
-						} else {
-							mkfile(t, "export OUT=${OUT}profile", filepath.Join(tmpDir, "launch", "app", ".profile"))
-						}
+						process.BuildpackID = "0.3/buildpack"
 					})
 
-					it("should source .profile", func() {
-						if err := launcher.Launch("/path/to/launcher", []string{"start"}); err != nil {
-							t.Fatal(err)
-						}
+					when("windows", func() {
+						it.Before(func() {
+							h.SkipIf(t, runtime.GOOS != "windows", "windows test")
+						})
 
-						stdout := rdfile(t, filepath.Join(tmpDir, "stdout"))
-						if len(stdout) == 0 {
-							stderr := rdfile(t, filepath.Join(tmpDir, "stderr"))
-							t.Fatalf("stdout was empty: stderr: %s\n", stderr)
-						}
-						if diff := cmp.Diff(strings.ReplaceAll(stdout, "\r\n", "\n"), "hi from app\nprof1,prof1start,prof2,prof2start,profile\n"); diff != "" {
-							t.Fatalf("syscall.Exec stdout did not match: (-got +want)\n%s\n", diff)
-						}
+						it("is not script", func() {
+							h.AssertNil(t, launcher.LaunchProcess("/path/to/launcher", process))
+							h.AssertEq(t, shell.nCalls, 1)
+
+							if shell.process.Script {
+								t.Fatalf("did not expect script process")
+							}
+						})
+					})
+
+					when("linux", func() {
+						it.Before(func() {
+							h.SkipIf(t, runtime.GOOS == "windows", "linux test")
+						})
+
+						it("is script", func() {
```

Comment:
I think I missed what it is that makes the process a script (it has args)... is that the default?

Reply:
I actually don't see what makes this test case different from https://github.com/buildpacks/lifecycle/pull/370/files#diff-3841e5f60214a915376943a789c0747cR492

Reply:
Hmmm the short `when`'s might be confusing in the code but I wanted to make the `go test` output a bit more readable on the command line

Should hopefully tell you it is a script if it has no args and is linux:
`LaunchProcess/Direct=false/buildpack-provided/buildpack_API_>=_0.4/has_no_args/linux/is_script`

The link above didn't take me to a different test.